Detectors/Vertexing/src/SVertexer.cxx

```diff
@@ -469,8 +501,9 @@ int SVertexer::checkCascades(float rv0, std::array<float, 3> pV0, float p2V0, in
     }
     int candC = 0;
     const auto& cascXYZ = fitterCasc.getPCACandidatePos(candC);
-    // make sure the cascade radius is smaller than that of the vertex
-    float dxc = cascXYZ[0] - pv.getX(), dyc = cascXYZ[1] - pv.getY(), dzc = cascXYZ[2] - pv.getZ(), r2casc = dxc * dxc + dyc * dyc;
+
+    // make sure the cascade radius is smaller than that of the mean vertex
+    float dxc = cascXYZ[0] - mMeanVertex.getX(), dyc = cascXYZ[1] - mMeanVertex.getY(), dzc = cascXYZ[2] - mMeanVertex.getZ(), r2casc = dxc * dxc + dyc * dyc;
```

Comment:
why using `mMeanVertex.getZ()` here? It can be many cm away from the real vertex.

Reply:
dzc was employed only for the causality cut. Now I moved the causality cut after the proper vertex computation.